index.js

```diff
@@ -59,237 +48,172 @@ Fs.readdirSync(Path.join(__dirname, 'helpers')).forEach(function (file) {
 * @param {getTranslationsCallback} callback - Callback when Assembler.getTranslations is done.
 */
 
-/**
-* Paper constructor. In addition to store settings and theme settings (configuration),
-* paper expects to be passed an assembler to gather all the templates required to render
-* the top level template.
-*
-* @param {Object} settings - Site settings
-* @param {Object} themeSettings - Theme settings (configuration)
-* @param {Object} assembler - Assembler with getTemplates and getTranslations methods.
-* @param {assemblerGetTemplates} assembler.getTemplates - Method to assemble templates
-* @param {assemblerGetTranslations} assembler.getTranslations - Method to assemble translations
-*/
 class Paper {
-    constructor(settings, themeSettings, assembler) {
-        this.handlebars = Handlebars.create();
-
-        this.handlebars.templates = {};
-        this.translator = null;
-        this.inject = {};
-        this.decorators = [];
-
-        this.settings = settings || {};
+    /**
+    * Paper constructor. In addition to store settings and theme settings (configuration),
+    * paper expects to be passed an assembler to gather all the templates required to render
+    * the top level template.
+    *
+    * @param {Object} siteSettings - Site settings
+    * @param {Object} themeSettings - Theme settings (configuration)
+    * @param {Object} assembler - Assembler with getTemplates and getTranslations methods.
+    * @param {String} rendererType - One of ['handlebars-v3', 'handlebars-v4']
+    * @param {assemblerGetTemplates} assembler.getTemplates - Method to assemble templates
+    * @param {assemblerGetTranslations} assembler.getTranslations - Method to assemble translations
+    */
+    constructor(siteSettings, themeSettings, assembler, rendererType) {
+        this.siteSettings = siteSettings || {};
         this.themeSettings = themeSettings || {};
         this.assembler = assembler || {};
-        this.contentServiceContext = {};
 
-        helpers.forEach(helper => helper(this));
-    }
+        // Build renderer based on type
+        switch(rendererType) {
```

Comment:
🍹 part of me wants to reduce this to something like:

```
this.renderer = new HandlebarsRenderer(this.siteSettings, this.themeSettings, rendererType.split(/-/).pop();
```

but being explicit is probably a better idea